src/ProductFilters/WidgetFilterDeliveryTime.php

```diff
@@ -88,17 +88,11 @@ public function widget($args, $instance) {
       else {
         $values = array_diff($values, [$delivery_time->term_id]);
       }
-      // Add filter values as query argument or remove parameter if empty.
-      // If the current value is selected the value will not be added.
-      if ($values) {
-        $link = add_query_arg('delivery_time', implode(',', $values));
-      }
-      else {
-        $link = remove_query_arg('delivery_time');
-      }
+      // Submit value as parameter.
+      $value = implode(',', $values);
       $chosen = in_array($delivery_time->term_id, $filter_values, TRUE);
       echo sprintf('<li class="%s">', $chosen ? $item_class . ' ' . $item_chosen_class : $item_class);
-      echo sprintf('<a rel="nofollow" href="%s">%s</a></li>', $link, $delivery_time->name);
+      echo sprintf('<form><a><button type="submit" class="product-filter-term" formmethod="GET" name="delivery_time" value="%s" style="border: none; background-color: #0000;">%s</button></a></form></li>', $value, $delivery_time->name);
```

Comment:
1. Why do you use an a tag in conjunction with a button if the goal is to remove the links? 🤔 
2. This is very hard to read so please put some formatting in :)
3. Don't use inline styles. Use classes instead

Putting a form element around looks like the only way if we don't want to do it with JS. If we do so, why don't you put the action and method on the form element itself? Also what happens, if you want to select multiple filters?

Reply:
Can we output the wrapping form element only once around all filters?